backend/app/crawler.py

```
@@ -278,21 +138,25 @@ async def discover_pages(
     parent_urls: Set[str] = None,
     all_internal_links: Set[str] = None,
     root_url: str = None,
-    root_task_id: str = None
+    root_task_id: str = None, # Keep for file naming consistency if needed, though job_id is primary now
```

Ruff (0.8.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)